calculator/src/test/java/com/market/banica/calculator/componentTests/CalculatorComponentIT.java

```diff
@@ -0,0 +1,279 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.json.JacksonTester;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.io.File;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ExtendWith({GrpcCleanupExtension.class})
+@ActiveProfiles("testIT")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+public class CalculatorComponentIT {
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private JMXServiceMBean jmxService;
+
+    @Autowired
+    private ProductBase productBase;
+
+    @Autowired
+    private TestConfigurationIT testConfigurationIT;
+
+    @SpyBean
+    @Autowired
+    private AuroraClientSideService auroraClientSideService;
+
+    @Value(value = "${client-id}")
+    private String clientId;
+
+    @Value(value = "${product-name}")
+    private String productName;
+
+    @Value(value = "${product-quantity}")
+    private int productQuantity;
+
+    @Value(value = "${product-price}")
+    private double price;
+
+    @Value(value = "${resource-timeout}")
+    private int timeout;
+
+    @Value(value = "${order-book-topic-prefix}")
+    private String orderBookTopicPrefix;
+
+    @Value(value = "${database-backup-url}")
+    private String databaseBackupUrl;
+
+    private Product product;
+
+    private Resources resources;
+    private Duration duration;
+
+    private AuroraServiceGrpc.AuroraServiceBlockingStub blockingStub;
+
+    private JacksonTester<RecipeDTO> jsonResponseRecipeDto;
+    private JacksonTester<List<Product>> jsonRequestProductList;
+    private JacksonTester<Product> jsonRequestProduct;
+
+    private String calculatorControllerGetRecipeUrl;
+    private String productControllerCreateProductUrl;
+
+    @BeforeEach
+    public void SetUp() {
+        JacksonTester.initFields(this, new ObjectMapper());
+        RestAssured.port = port;
+
+        calculatorControllerGetRecipeUrl =
+                "calculator/" + clientId + "/" + productName + "/" + productQuantity;
+        productControllerCreateProductUrl = "product";
+
+        product = new Product();
```

Comment:
it would be better to extract it in a separate method

Reply:
Done